nimble/host/include/host/ble_gap.h

```diff
@@ -292,6 +295,19 @@ struct hci_conn_update;
 
 /** @} */
 
+/**
+ * @defgroup GAP Authorize event possible responses.
+ * @{
+ */
+
+/** GAP Authorize event response: accept */
+#define BLE_GAP_AUTHORIZE_ACCEPT            1
+
+/** GAP Authorize event response: reject */
+#define BLE_GAP_AUTHORIZE_REJECT            2
```

Comment:
authorization can be done using return value from event callback

Reply:
I see two problems with that.

1. The return values are normally used for error codes, I guess maybe returning BLE_HS_EREJECT will work for rejecting.
2. The default return code in gap_event in the apps is 0. So if the user does not handle the event, the app will return 0, which will be interpreted as "ACCEPT".

Reply:
1. just return non-zero and this means reject
2. that's expected behavior, otherwise apps that do not have this event handled would reject all requests which is not what we want